node/network/bridge/src/validator_discovery.rs

```diff
@@ -135,6 +140,13 @@ fn peer_id_from_multiaddr(addr: &Multiaddr) -> Option<PeerId> {
 }
 
 pub(super) struct Service<N, AD> {
+	state: HashMap<PeerSet, StatePerPeerSet>,
```

Comment:
Nit: Looking at this code I can't help but notice that this should be a dedicated struct. A hashmap is a bit heavy for a binary key and it brings a bit of a hassle with these `.or_default`.